### ISSUE TITLE
A question about LSTM's input dimensions

### ISSUE THREAD

**Issue by narrowview** [fchollet/keras#1881]
I'm a novice just using Keras' lstm.
I make up my lstm model and substitution the input.
There is a TypeError:('Bad input argument to theano function with name "/usr/local/lib/python2.7/dist-packages/keras/backend/theano_backend.py:380" at index 0(0-based)','Wrong number of dimensions:expected 3,got 1 with shape(128,).')
Here is my code(my x_train is 1 dimension in the form of numpy.float and my y_train is 8 dimension in the form of numpy.array)     

  model = Sequential()
  model.add(LSTM(8,input_dim = 1,activation = 'tanh',inner_activation =     'hard_sigmoid',return_sequences = True))
  model.add(Dropout(0.2))
  model.add(LSTM(16,activation = 'tanh',inner_activation = 'hard_sigmoid',return_sequences = True))
  model.add(Dropout(0.2))
  model.add(LSTM(16,activation = 'tanh',inner_activation = 'hard_sigmoid',return_sequences = True))
  model.add(Dropout(0.2))
  model.add(TimeDistributedDense(12))
  model.add(Activation('sigmoid'))

  model.compile(loss = 'mean_squared_error',optimizer = 'sgd')
  model.fit(x_train,y_train,nb_epoch = 3,verbose = 1)

And I have another question about Sequence classification with LSTM in Keras Documentation's Examples.
The question is what is the use of "model.add(Embedding(max_features, 256, input_length=maxlen))".
The Embedding is sum or mul or others?

**Comment by ymcui** [comment]
According to your model,
the input should be in shape `(batch_size, length_of_sequence, 1)`
and the output should be in shape `(batch_size, length_of_sequence, 12)`
so, check your x_train and y_train to see if they match the shape of above.

`Embedding` model is to convert indices(such as word index in a vocabulary) into a dense representation.

**Comment by narrowview** [comment]
Thanks ymcui
